Storage/ThirdParty/SwiftData.swift

```diff
@@ -113,16 +119,18 @@ public class SwiftData {
             conn = ConcreteSQLiteDBConnection(filename: filename, flags: flags.toSQL(), key: self.key, prevKey: self.prevKey)
         }
 
-        // Initialize in case we encounter a nil connection when we go to run the dispatch blocks
-        let failed = FailedSQLiteDBConnection()
         let queue = self.sharedConnectionQueue
-        let noConnection = NSError(domain: "mozilla", code: 0, userInfo: [NSLocalizedDescriptionKey: "Could not create a connection"])
-
         if synchronous {
             var error: NSError? = nil
             dispatch_sync(queue) {
+                /**
+                 * By the time this dispatch block runs, it is possible the user has backgrounded the app
+                 * and the connection has been dealloc'ed since we last grabbed the reference
```

Comment:
> some consistent collection of writes is performed — e.g., none of A, B, C, or only C, or all of A, B, C.

At it's core, each write operation is queued up on the `sharedConnectionQueue` which is a serial dispatch queue. In either the synchronous or async cases of `withConnection`, the writes have guaranteed ordering since that's what a serial dispatch queue provides us. Therefore, `long write X`, `write A`, `write B` all execute in this order and before the `forceClose` call. Since queue is serial and `forceClose` is called after `write C`, we know that the weak ref'ed connection we use in those writes will hold.